airbyte-cdk/java/airbyte-cdk/db-sources/src/testFixtures/java/io/airbyte/cdk/testutils/PostgresTestDatabase.java

```diff
@@ -242,38 +275,45 @@ public void withWalLevelLogical() {
+        // reconfigure postgres
         "echo \"ssl = on\" >> /var/lib/postgresql/data/postgresql.conf",
-        "echo \"ssl_cert_file = '/etc/ssl/private/server.crt'\" >> /var/lib/postgresql/data/postgresql.conf",
-        "echo \"ssl_key_file = '/etc/ssl/private/server.key'\" >> /var/lib/postgresql/data/postgresql.conf",
-        "echo \"ssl_ca_file = '/etc/ssl/private/ca.crt'\" >> /var/lib/postgresql/data/postgresql.conf",
-        "mkdir root/.postgresql",
-        "echo \"hostssl    all    all    127.0.0.1/32    cert clientcert=verify-full\" >> /var/lib/postgresql/data/pg_hba.conf",
-        "openssl ecparam -name prime256v1 -genkey -noout -out client.key",
-        "openssl req -new -sha256 -key client.key -out client.csr -subj \"/CN=postgres\"",
-        "openssl x509 -req -in client.csr -CA ca.crt -CAkey ca.key -CAcreateserial -out client.crt -days 365 -sha256",
-        "cp client.crt ~/.postgresql/postgresql.crt",
-        "cp client.key ~/.postgresql/postgresql.key",
-        "chmod 0600 ~/.postgresql/postgresql.crt ~/.postgresql/postgresql.key",
-        "cp ca.crt root/.postgresql/ca.crt",
-        "chown postgres:postgres ~/.postgresql/ca.crt",
-        "psql -U test -c \"SELECT pg_reload_conf();\"",
+        "echo \"ssl_cert_file = '/var/lib/postgresql/certs/server.crt'\" >> /var/lib/postgresql/data/postgresql.conf",
+        "echo \"ssl_key_file = '/var/lib/postgresql/certs/server.key'\" >> /var/lib/postgresql/data/postgresql.conf",
+        "echo \"ssl_ca_file = '/var/lib/postgresql/certs/ca.crt'\" >> /var/lib/postgresql/data/postgresql.conf",
+        // Here, we reset pg_hba to not accept any connection except locals.
+        "echo \"local all test all trust\" > /var/lib/postgresql/data/pg_hba.conf",
+        // Then we add SSL-only with full certification to the user used by all network connections.
+        "echo \"hostssl all " + sharedContainer.getUsername() + " all cert\" >> /var/lib/postgresql/data/pg_hba.conf",
+        // finally, create client key and certificate, both verified by the CA
+        "openssl ecparam -name prime256v1 -genkey -noout -out /var/lib/postgresql/certs/client.key",
+        "openssl req -new -sha256 -key /var/lib/postgresql/certs/client.key -out /var/lib/postgresql/certs/client.csr -subj \"/CN="
+            + sharedContainer.getUsername() + "\"",
+        "openssl x509 -req -in /var/lib/postgresql/certs/client.csr -CA /var/lib/postgresql/certs/ca.crt -CAkey /var/lib/postgresql/certs/ca.key " +
+            "-CAcreateserial -out /var/lib/postgresql/certs/client.crt -days 365 -sha256",
+        // make everything accessible by postgres only, as required by the postgres doc at
+        // https://www.postgresql.org/docs/16/ssl-tcp.html
+        "chmod 0600 /var/lib/postgresql/certs/*",
+        "chown postgres:postgres /var/lib/postgresql/certs/*",
+        // reload config and pg_hba.
+        "psql -U test -c \"SELECT pg_reload_conf();\""
```

Comment:
I'll take you at your word that this stuff works. If the CI is happy then I'm happy.